src/Pos/Block/Logic/Internal.hs

```diff
@@ -79,17 +82,29 @@ type MonadBlockBase ssc ctx m
        -- Needed by some components.
        , MonadGState m
        -- This constraints define block components' global logic.
-       , HasLens TxpGlobalSettings ctx TxpGlobalSettings
        , HasLens LrcContext ctx LrcContext
+       , HasLens TxpGlobalSettings ctx TxpGlobalSettings
        , SscGStateClass ssc
+       , HasLens GenesisStakeholders ctx GenesisStakeholders
+       , MonadDelegation ctx m -- TODO: split dlgApplyBlocks blocks
```

Comment:
because we introduce `normalize` (normalisation of mempool as separate function), it's not part of `applyBlocksUnsafe` anymore